core/src/main/java/io/aiven/klaw/config/ManageDatabase.java

```diff
@@ -185,20 +185,31 @@ private void loadStaticDataToDb() {
           defaultDataService.getTeam(KwConstants.DEFAULT_TENANT_ID, stagingTeam));
     }
 
-    // add user
-    UserInfo userExists = handleDbRequests.getUsersInfo(superAdminDefaultUserName);
-    if (userExists == null) {
-      handleDbRequests.addNewUser(
-          defaultDataService.getUser(
-              KwConstants.DEFAULT_TENANT_ID,
-              superAdminDefaultPwd,
-              "SUPERADMIN",
-              handleDbRequests
-                  .selectTeamDetailsFromName(infraTeam, KwConstants.DEFAULT_TENANT_ID)
-                  .getTeamId(),
-              kwAdminMailId,
-              superAdminDefaultUserName,
-              encryptorSecretKey));
+    // check if default superadmin is configured or blank.
+    // in case of blank, there should be atleast one other user with SUPERADMIN role
+    if (superAdminDefaultUserName.isBlank()) {
+      if (!validateUsersBeforeAdding()) {
```

Comment:
I would do the same for the default password and fail if the config doesn't have any value.

Reply:
agree, done.